test/gas/lib/isGasExpenseCorrect.js

```diff
@@ -0,0 +1,42 @@
+const assert = require("assert");
+const to = require("../../../lib/utils/to.js");
+
+const isGasExpenseCorrect = async(expectedGasPrice, setGasPriceOnTransaction = false, web3, accounts) => {
+  const transferAmount = web3.utils.toBN(web3.utils.toWei("5", "finney"));
+  const expectedGasPriceBN = web3.utils.toBN(expectedGasPrice);
+
+  const balance = await web3.eth.getBalance(accounts[0]);
+  const initialBalance = await web3.utils.toBN(balance);
+
+  const params = {
+    from: accounts[0],
+    to: accounts[1],
+    value: transferAmount
+  };
+
+  if (setGasPriceOnTransaction) {
+    params.gasPrice = expectedGasPriceBN;
+  }
+
+  const receipt = await web3.eth.sendTransaction(params);
```

Comment:
The name of the function (the "is" prefix, specifically) makes me think it's "pure" - meaning it has no side effects, but we're changing state on this line

I think we're asking too much of this function, however. I'd rather see it get broken down into smaller chunks instead of having its name changed

Reply:
Also re: encapsulation - based on the name I'd expect this function to contain the logic that validates a given transaction's gas usage is correct, but we depend on the caller to do that.

That said, I don't think we should be rewriting gas calculation logic into our test suite. I take this as yet more indication that we're asking too much of this function

Reply:
Can change the name of the function and break up the function.

Reply:
This file was renamed.